xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/transactions/Memo.java

```diff
@@ -20,7 +25,13 @@
 @Value.Immutable
 @JsonSerialize(as = ImmutableMemo.class)
 @JsonDeserialize(as = ImmutableMemo.class)
-public interface Memo {
+public abstract class Memo {
+
+  static ImmutableMemo.Builder withPlainText(final String plainText) {
+    return builder()
+      .memoFormat(BaseEncoding.base16().encode("text/plain".getBytes(StandardCharsets.UTF_8)))
+      .memoData(BaseEncoding.base16().encode(plainText.getBytes(StandardCharsets.UTF_8)));
+  }
```

Comment:
Should we also include a static constructor that lets you pass in a plaintext value for `MemoType`?

Reply:
Not sure really - I didn't want to go overboard here except to support a plain-text memo. `MemoType` is meant to express a Link-Relation and I feel like anyone wanting to do that will need to encoded their URL as HEX manually -- this PR is really just meant to provide a simple utility to make plain-text easier, and to perhaps illustrate to devs how this should work.

Reply:
Not opposed to this for now. We can always add another constructor later on.

> I feel like anyone wanting to do that will need to encoded their URL as HEX manually 

Shouldn't we provide a utility for them to encode URLs to hex automatically? Is there something about hex encoding URLs that is user specific?

Reply:
@nkramer44 and I decided to leave this out for now.